internal/git/gitsync_processor.go

```diff
@@ -157,8 +185,228 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1.GitSync, rest
 			return err
 		}
 	} else {
-		// TODO: monitoring with intervals
-		logger.Debug("monitoring with intervals")
+		ticker := time.NewTicker(timeInterval * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				patchedResources, recentHash, err := CheckForRepoUpdates(ctx, r, repo, &gitSync.Status, namespace)
+				if err != nil {
+					return err
+				}
+
+				err = updateCommitStatus(ctx, k8Client, namespacedName, recentHash, repo, logger)
+				if err != nil {
+					return err
+				}
+
+				err = ApplyPatchToResources(patchedResources, client)
+				if err != nil {
+					return err
+				}
+
+			case <-ctx.Done():
+				logger.Debug("Context canceled, stopping updates check")
+				return nil
+			}
+		}
+	}
+	return nil
+}
+
+// ApplyPatchToResources applies changes to Kubernetes resources based on the provided PatchedResource.
+// It uses a Kubernetes client to apply changes to each resource and handles creation and deletion of resources as needed.
+
+func ApplyPatchToResources(patchedResources PatchedResource, client kubernetes.Client) error {
+	for key, afterValue := range patchedResources.After {
+		namespace := strings.Split(key, "/")[0]
+		if beforeValue, ok := patchedResources.Before[key]; ok {
+			if beforeValue != afterValue {
+				err := client.ApplyResource([]byte(afterValue), namespace)
+				if err != nil {
+					return errors.New("failed to apply resource: " + err.Error())
+				}
+			}
+		} else {
+			// Handle newly added resources
+			err := client.ApplyResource([]byte(afterValue), namespace)
+			if err != nil {
+				return errors.New("failed to apply new resource: " + err.Error())
+			}
+		}
+	}
+
+	// Handle deleted resources
+	for key, beforeValue := range patchedResources.Before {
+		resource, err := yamlUnmarshal(beforeValue)
+		if err != nil {
+			return errors.New("failed to unmarshal YAML: " + err.Error())
+		}
+		if _, ok := patchedResources.After[key]; !ok {
+			err := client.DeleteResource(resource.Kind, resource.Metadata.Name, resource.Metadata.Namespace, metav1.DeleteOptions{})
+			if err != nil {
+				return errors.New("failed to delete resource: " + err.Error())
+			}
+		}
+	}
+	return nil
+}
+
+// this check for file changes in the repo by comparing the old commit  hash with new commit hash and returns the recent hash with PatchedResources map
+
+func CheckForRepoUpdates(ctx context.Context, r *git.Repository, repo *v1.RepositoryPath, status *v1.GitSyncStatus, defaultNameSpace string) (PatchedResource, string, error) {
+	var patchedResources PatchedResource
+	var recentHash string
+	logger := logging.FromContext(ctx)
```

Comment:
```suggestion
	logger := logging.FromContext(ctx).With("GitSync name", gitSync.Name,
		"RepositoryPath Name", repo.Name)
```

Reply:
this will make it so that our logger will automatically include this information so you don't need to include it in each log line below

Reply:
gitsync is not available in this ,do you want me to pass as an argument?